stdlib/strscan.rb

```diff
@@ -297,6 +297,10 @@ def unscan
 
 private
   def anchor(pattern)
-    `new RegExp('^(?:' + pattern.toString().substr(1, pattern.toString().length - 2) + ')')`
+    %x{
+      var flags = pattern.toString().match(/\/([^\/]+)$/)
+      flags = flags && flags[1] || undefined
+      return new RegExp('^(?:' + pattern.source + ')', flags)
```

Comment:
Can you add the final semicolon to all the JS lines?

Trailing semicolons wars aside, it should fix the specs because opal detects its presence and adds a leading return if its missing.

eg.

```rb
def anchor(pattern)
  %x{
    var flags = pattern.toString().match(/\/([^\/]+)$/)
    flags = flags && flags[1] || undefined
    return new RegExp('^(?:' + pattern.source + ')', flags)
  }
end
```

through `| opal -cEO` generates:

```js
/* Generated by Opal 0.11.0.dev */
(function(Opal) {
  var TMP_anchor_1, self = Opal.top, $scope = Opal, $scopes = [Opal], nil = Opal.nil, $breaker = Opal.breaker, $slice = Opal.slice;

  return (Opal.defn(Opal.Object, '$anchor', TMP_anchor_1 = function $$anchor(pattern) {
    var self = this;

    return 
    var flags = pattern.toString().match(/\/([^\/]+)$/)
    flags = flags && flags[1] || undefined
    return new RegExp('^(?:' + pattern.source + ')', flags)
  
  }, TMP_anchor_1.$$arity = 1), nil) && 'anchor'
})(Opal);
```